shell/platform/darwin/macos/framework/Headers/FlutterPluginRegistrarMacOS.h

```diff
@@ -36,11 +36,30 @@ FLUTTER_DARWIN_EXPORT
 @property(nonnull, readonly) id<FlutterTextureRegistry> textures;
 
 /**
- * The default view displaying Flutter content.
+ * The view displaying Flutter content.
  *
- * This method may return |nil|, for instance in a headless environment.
+ * Currently Flutter only supports one view, and this is the view.
  *
- * The default view is a special view operated by single-view APIs.
+ * Flutter plans to support multiple views in the future. Although single-view
+ * APIs will eventually be replaced by their multi-view variants, during the
+ * deprecation period, the single-view APIs will coexist with and work with the
+ * multi-view APIs as if the other views don't exist. To achieve this,
+ * all behaviors of "the single view" (which is called "the implicit view") are
+ * preserved, allowing legacy single-view APIs to continue working, while
+ * new-style views created by new ways must be operated by the upcoming
+ * multi-view APIs.
+ *
+ * Plugins written for a single view can keep operating on this view and expect
+ * unchanged behavior for the implicit view. This includes that:
+ *
+ *  - The first view controller attached to the engine will be linked to the
+ *    implicit view.
+ *  - Single-view Flutter APIs will operate the implicit view.
+ *
+ * This method may return |nil| if the view is not assigned. In single-view
+ * apps, this means that the app is running headlessly. In multi-view apps,
+ * this means that the compatible mode is disabled, or that the compatible mode
```

Comment:
We don't define what "compatible mode" is anywhere and if we can avoid it I wouldn't introduce it as a new concept. Can we instead explain this in terms of existing concepts like the implicit view? E.g. maybe we can just say that this being null means the app has been configured to run without an implicit view. Or something like that.